### ISSUE TITLE
Techniques stuck in "DETONATED" when resource does not exist

### ISSUE THREAD

**Issue by lauraf-c42** [DataDog/stratus-red-team#167]
# What is not working?
The Code42 SecOps team has identified two techniques which can get stuck in the DETONATED state and fail to CLEANUP or REVERT (even using force flags). 

* aws.exfiltration.s3-backdoor-bucket-policy
* aws.defense-evasion.cloudtrail-lifecycle-rule

Techniques enter this state when the AWS infrastructure used to run the test has been deleted from the environment, but the local Stratus still has the state as DETONATED.

Neither the REVERT nor CELANUP with force flags were successful in restoring to a WARM or COLD state. 

It appears that the CLEANUP and REVERT attempt to clean up the AWS infrastructure by making API calls to delete the resources that had been created when the technique was detonated. This makes sense. However, upon receipt of a 404 error, it simply prints the error and leaves the technique in the detonated state.

## What SHOULD happen?
Upon receipt of a 404 error during CLEANUP (and/or REVERT), particularly with `--force`:
* Execution should continue to refresh the state file to indicate the resource no longer exists
* REVERT (--force)
    * Revert to WARM state, ready to run again (probably need checks to confirm warmup infrastructure exists)
* CLEANUP (--force)
    * Revert to COLD state

# System Details

| OS Version | Stratus Version 
| - | - |
| Mac OS X | 2.0.0 | 

# Full output?
```
$ stratus cleanup aws.defense-evasion.cloudtrail-lifecycle-rule --force
2022/09/06 11:51:53 Cleaning up aws.defense-evasion.cloudtrail-lifecycle-rule
2022/09/06 11:51:53 Reverting detonation of technique aws.defense-evasion.cloudtrail-lifecycle-rule
2022/09/06 11:51:53 Reverting S3 Lifecycle Rules on CloudTrail S3 bucket my-cloudtrail-bucket-gmctuebrbigvfczn
2022/09/06 11:51:54 unable to revert detonation of aws.defense-evasion.cloudtrail-lifecycle-rule: unable to revert detonation of aws.defense-evasion.cloudtrail-lifecycle-rule: unable to revert S3 Lifecycle Policy Rule: operation error S3: DeleteBucketLifecycle, https response error StatusCode: 404, RequestID: request_id, HostID: host_id, api error NoSuchBucket: The specified bucket does not exist
+-----------------------------------------------+------------------------------------------------------+-----------+
| ID                                            | NAME                                                 | STATUS    |
+-----------------------------------------------+------------------------------------------------------+-----------+
| aws.defense-evasion.cloudtrail-lifecycle-rule | CloudTrail Logs Impairment Through S3 Lifecycle Rule | DETONATED |
+-----------------------------------------------+------------------------------------------------------+-----------+
```

# Files in $HOME/.stratus-red-team?
If these are needed, I can get my local in the correct state to recreate them. 

# Workaround - Return to COLD state
Manually deleting the files at `$HOME/.stratus-red-team/technique` restored to a COLD state.

**Comment by christophetd** [comment]
Confirmed the behavior for `aws.defense-evasion.cloudtrail-lifecycle-rule`:

```
$ stratus detonate aws.defense-evasion.cloudtrail-lifecycle-rule
$ aws s3 rm --recursive s3://...bucketname
$ aws s3api delete-bucket --bucket ...bucketname
$ stratus cleanup aws.defense-evasion.cloudtrail-lifecycle-rule
2022/09/13 08:57:22 unable to revert detonation of aws.defense-evasion.cloudtrail-lifecycle-rule: unable to revert detonation of aws.defense-evasion.cloudtrail-lifecycle-rule: unable to revert S3 Lifecycle Policy Rule: operation error S3: DeleteBucketLifecycle, https response error StatusCode: 404, RequestID: D97KEP6ZT99YFJN3, HostID: zZlH6Zezduyw9aEJooIaZIScoFTtswHQyTqllAJ+p78/4cqLVgkOkIdfFgUPKmp+Gb5prZHTUao=, api error NoSuchBucket: The specified bucket does not exist
```

**Comment by christophetd** [comment]
Confirmed the behavior for `aws.exfiltration.s3-backdoor-bucket-policy` similarly.

**Comment by christophetd** [comment]
Thanks for reporting @lauraf-c42! I implemented the following behavior in this branch (https://github.com/DataDog/stratus-red-team/pull/169), can you give it a try and confirm it matches your expectations?

When cleaning up with the `--force` flag, if `revert` returns an error, it's treated as a warning only.

I attached a build of the branch for your convenience! https://drive.google.com/file/d/1XqDvFS7CZrQyiNNSgprKKB8pRvz5P1bs/view?usp=sharing (d2330d16203ba64c89ea0b6251fa40b7461deff0540a37d1e9e7fe6ea6f666a8)

Thanks again!

**Comment by lauraf-c42** [comment]
Thanks @christophetd ! I cloned the branch and tested in my local, and it looks like the extra enforcement you applied worked like a charm! (if you'll excuse my pun :D )

✅ Looks great to me!

**Comment by christophetd** [comment]
Amazing! I'll get it merged right away, appreciate the report!

**Comment by christophetd** [comment]
Now released as part of v2.4.1: https://github.com/DataDog/stratus-red-team/releases/tag/v2.4.1
